app/project_type/git.py

```diff
@@ -41,7 +41,9 @@ def _generate_path_from_repo_url(base_sys_path, url):
         url_folder_path_parts = url_full_path_parts[:-1]
         repo_directory = os.path.join(
             base_sys_path,
-            url_components.netloc.replace(':', ''),  # remove colons from netloc (e.g. turn example:8000 to example8000)
+            # Remove colons from netloc/scheme (e.g. turn example:8000 to example8000).
+            # If netloc is empty the url is malformed -- use scheme instead.
+            (url_components.netloc or url_components.scheme).replace(':', ''),
```

Comment:
Maybe create an intermediate variable that holds the result of (netloc or scheme)? It might make this a bit easier to read. I think the comment is good though.